src/components/ProfileMatch/index.tsx

```diff
@@ -0,0 +1,166 @@
+  Listing,
+  Profile,
+} from '@/types/schema';
+import { useProfile } from '@/utils/ProfileProvider';
+import {
+  timestampStringToDate,
+  formatTimestamp,
+  parseLanguageList,
+} from '@/utils/helpers';
+import CONFIG from '@/lib/configs';
+import COLORS from '@/styles/colors';
+import Icon from '../Icon';
+
+interface MatchField<T extends Listing> {
+  getMatch: (data: T, profileData: Profile) => boolean | undefined;
+  getText: (
+    data: T,
+    profileData: Profile,
+    match: boolean | undefined,
+  ) => string;
+}
+
+const timeCommitmentMatch: MatchField<CaseListing> = {
+  getMatch: (data, profileData) =>
+    data.hours_per_week
+      ? profileData.hours_per_month > data.hours_per_week * 4
+      : undefined,
+  getText: (data, profileData, match) =>
+    data.hours_per_week
+      ? `Your time commitment (${profileData.hours_per_month} hours/month) ${
+          match ? 'meeets' : 'does not meet'
+        } the minimum for this case (${data.hours_per_week * 4} hours/month).`
+      : 'No information available about Time Commitment',
+};
+
+const locationMatch: MatchField<CaseListing> = {
+  getMatch: (data, profileData) =>
+    data.client_location
+      ? data.client_location === profileData.location
+      : undefined,
+  getText: (data, profileData, match) => match === undefined
+      ? 'No information available about Client Location'
+      : `This client is ${match ? '' : 'not '}in your city.`,
+};
+
+const startDateMatch: MatchField<
+  CaseListing | LimitedCaseAssignment | DocumentTranslation
+> = {
+  getMatch: (data, profileData) => {
+    const date = data.listing_type === 'CASE' ? data.upcoming_date : data.deadline; 
+    return date
+        ? timestampStringToDate(profileData.start_date) <
+            timestampStringToDate(date)
+        : undefined;
+  },
+  getText: (data, profileData, match) =>
+    match === undefined
+      ? `No information available about the ${
+          data.listing_type === 'CASE'
+            ? 'Next Court Hearing/Filing Date'
+            : 'Deadline'
+        }`
+      : `The ${
+          data.listing_type === 'CASE'
+            ? 'next court hearing/filing date'
+            : 'deadline'
+        } (${formatTimestamp(
+          data.listing_type === 'CASE' ? data.upcoming_date : data.deadline,
+        )}) is ${
+          match ? 'after' : 'before'
+        } your first available start date (${formatTimestamp(
+          profileData.start_date,
+        )}).`,
+};
+
+const matchIcon = (match: boolean | undefined) => {
+  if (match === undefined) {
+    return <Icon type="gray_dot" />; // change to yellowExclaimation
+  }
+  if (match) {
+    return <Icon type="green_check" />; // rename to greenCheck later
+  }
+  return <Icon type="redCross" />;
+};
+
+export default function ProfileMatch({
+  listingData,
+  interpretation = false,
+}: {
+  listingData: Listing;
+  interpretation?: boolean;
+}) {
+  const profile = useProfile();
+  const matchedLanguages = profile
```

Comment:
should this be useMemo in case profile data changes?

Reply:
yep! i would also recommend destructuring so it only changes when the field changes
i.e.
```typescript
const { languages } = profile;
const matchedLanguages = useMemo(() => ..., [languages])
```